### ISSUE TITLE
Evaluate Pump

### ISSUE THREAD

**Issue by PRemmen** [RWTH-EBC/AixLib#184]
Evaluate how to proceed with Pump

Advantages/Disadvantages to have it in parallel to Annex60 model (FlowControlled_dp/_m_flow)?

**Comment by AnaConstantin** [comment]
- [ ] Make list of advantages and disadvantages of Pump model in Annex60 and Pump Model in HVAC

- [ ] Decide on one (unified) model

- [ ] Build model

- [ ] Add records

- [ ] Build examples

- [ ] Document model, record, examples

**Comment by MBaranskiEBC** [comment]
@pema1: could your pump model be included/contribute to the unified model?

**Comment by AnaConstantin** [comment]
I'm open to any suggestions on how the model can be enhanced. For me it is only important to bring the advantages of our HVAC model (mainly usage of records for characteristics and different established control strategies) to our AixLib model.

**Comment by pema1** [comment]
@MBaranskiEBC, our current model is based on a rather complex model from @AnaConstantin. We currently simplify it and can hopefully improve it in some details. I will then have a look if the this model could benefit from our work.

**Comment by AnaConstantin** [comment]
After reviewing our HVAC models and the Annex60/IBPSA-Models I have identified the following differences:
1. Our models use tabels with (Vflow, Head) for different nominal speeds (according to manufacturer data) and a given control strategy to calculate the needed pressure drop. Power and efficiency calculations are done using polynoms, which are the result of curve fittings in MATLAB over manufacturer data   (Vflow, Head, n) and (Vflow, Power, n).
2. The Annex60 / IBPSA models (Fluid-Movers) focus on calculating the electrical power (and pressure drop when input = rotational speed) of the pump using sets of data (Vflow, Head) and (Vflow,Power) for a nominal speed using manufacturer data. It's a cool implementation. This power can be than added to the medium (for fans maybe more relevant than for pumps). However no control strategies for pumps / fans are implemented separately and no power limitation according to manufacturer data is included explicitly in the model.  
I propose the following:
1. Separate control and physical model in our HVAC models and make a control block with the relevant control strategies and introduce them in AixLib
2. As a physical model the choice can be made between the simple implementation we have in HVAC or the more complex in IBPSA / Annex60.
3. I like that manufacturer data in records to resemble manufacturer data as found in the data sheets (especially measurement units). We need to transfor the information from our records in the information needed in the IPBSA / Annex60 records. I will think of a solution
4. I would like to test if the implementation for power calculations from IBPSA / Annex60 is quicker and more stable than using polynoms obtained from curve fitting with MATLAB. It is clearly a lot easier to use, as the polynoms do not have to be calculated.

**Comment by pema1** [comment]
Hello @AnaConstantin,

We currently experience one rather important problem with our pump model (and probbaly with the Annex60 models also) in that it cannot consider other pumps in a network properly. If you have two pumps in a row for example the stronger one will push through the smaller. The amount of water can be considerable in real life but that is often not considered when designing a system. The effect can be big. 

If one pump stops for example and the other still runs this effect becomes visible. In our case the mass flow would go to zero if I am not mistaken but in reality it would not. This can lead to awkward problems with hydraulics and control which we cannot predict at the moment. However, it would nice if we could see this in a simulation.

I am not sure how to implement this best. Maybe manipulating the pressure drop in some way so as to achieve an overlay that will consider other forces too. When the pump comes to a halt it should work only as a hydraulic resistance like most other hydraulic components.

**Comment by Mathadon** [comment]
@pema1 what pump do you use? And what pressure drops are there downstream of the pumps. The pumps are not really designed to be used in series but I think it should be possible if your overall system makes sense.

**Comment by AnaConstantin** [comment]
@pema1 my suggestion is to try and implement a resistance over the on/off block in our HVAC model. Now it is a smooth between zero and the current head value of the pump, when the pump turns off. We can just put another value for zero for a quick test. It should definetly work when there is another pump working in the system and the second pump turns off. I'm curios how it will work with just one pump in the system.
@Mathadon thanks for joining in on the conversation, as you have worked on the pump model from IBPSA/Annex60. Do you agree with my assesment of the model, or am I missing something?

**Comment by Mathadon** [comment]
I'm not sure what the system looks like so it's hard to tell :)

**Comment by AnaConstantin** [comment]
@Mathadon I meant the assement of the pump models in the IPBSE/Annex60 library, the ones in the Fluid.Movers -Package.

**Comment by pema1** [comment]
@Mathadon I am using an internal model developed at our institute. It is not public yet. I will ask for permission to share it with you for discussion. The hydraulic resistances in the network are mostly simple pressure drops in pipes and junctions of the form dp=f(C, m_flow), where C is a resistance factor. Valves from MSL are also included. I try to use only MSL or AixLib components if possible.

@AnaConstantin, I think your idea is not sufficient as the resistance effect of a pump will not suddenly appear when rpm==0 but increase from the point when the large pump will provide a higher pressure increase than the small - I believe. I should ever increase until the small pump comes to a complete halt which is when the resistance will be biggest. An even more extreme situation would be when the pumps would work against each other. Then the small pump would experience a reverse flow even when it is turned on. Would be nice if this could be modeled in a universal fashion.

**Comment by mlauster** [comment]
@pema1, what to do on this issue? Is somebody working on that? Is still urgent? Shall we postpone it?

**Comment by pema1** [comment]
Hello @mlauster, @AnaConstantin, @Mathadon,

the model in the 'Zugabe' project is finished now. Regarding @AnaConstantin's suggestions:

* We have two models, one that uses a pressure difference signal as input and one that uses pump speed as input.
* The first will directly force the pressure difference onto the system resembling an ideal pump. Our partner wanted such a model variant as they hope the simplification will be useful in certain situations. This model will limit the pump delivery head and volume flow rate to the maximum and minumm pump curves given in a lookup table (as in the current model).
* The latter model will use a lookup table for max. and min. allowed pump speed for a given volume flow rate. Pump delivery head and power consumption will be calculated by a 2D polynomial (volume flow rate, pump speed) with an arbitrary number of terms. The polynomial coefficients are given in arrays.

1. Separate control and physical model
 For both pump models there is a base model without controller that can be used as is and also a model that integrates a controller block. For both models we have a dp_var controller that can also be used to mimic a dp_const strategy (if H_0==H_design).
3. manufacturer data in records
All pump specific parameters are provided by a record. For example the coefficient tables for pump delivery head and pump power. I developed a python script that automatically generates a modelica record from a table of measurement data. The record will also include a simple documentation with graphics that show the measured and fitted pump curves.
4. power calculations from IBPSA / Annex60 (no polynomes)
The model uses polynomes. We had the measurement data to do this. Also the data sheets from the manufacturers have those data. But it would probably be possible to change this. Higher order polynomes showed problems with initialization of the solver in complex simulations. However, power currently is only an output variable and should not affect initialization.

* As you can see this model also does not compute a force that will be imposed onto the fluid as to allow for inline pumps or inverse flow. One thing that came to my mind is how do we calculate or guess the hydraulic resitance of the pump at speed 0 rpm?

I would like to put this model in a development branch so you can test it. But will need to generate some measurement data and a respective record as I cannot use the partner's measurment data.

**Comment by mlauster** [comment]
Covered by #563
